backend/src/test/java/heartbeat/service/report/ReportServiceTest.java

```diff
@@ -104,4 +101,25 @@ void ShouldInitializeMetricsDataCompletedInHandlerWithPreOneWhenPreOneExisted()
 		verify(generateReporterService, never()).generateDoraReport(request);
 	}
 
+	@Test
+	void ShouldInitializeMetricsDataCompletedInHandlerWhenRequestMetricsExist() throws InterruptedException {
+		GenerateReportRequest request = GenerateReportRequest.builder()
+			.csvTimeStamp("csvTimeStamp")
+			.metrics(List.of("velocity", "lead time for changes"))
+			.build();
+		when(asyncMetricsDataHandler.getMetricsDataCompleted(any()))
+			.thenReturn(MetricsDataCompleted.builder().pipelineMetricsCompleted(true).build());
+		doAnswer(invocation -> null).when(generateReporterService).generateBoardReport(request);
+		reportService.generateReportByType(request, MetricType.BOARD);
```

Comment:
new line for `given when then`

Reply:
when it's too long to pass e2e.

Reply:
Sorry, I didn't get it. Change it right away

Reply:
done